src/ufs.rs

```diff
@@ -432,7 +432,15 @@ impl Filesystem for Ufs {
 					let len = ino.size as usize;
 					Ok(link[0..len].to_vec())
 				}
-				_ => Err(IoError::new(ErrorKind::Unsupported, "TODO: long links")),
+				InodeData::Blocks { .. } => {
+					// TODO: this has to be tested for other configurations, such as 4K/4K
+					assert!(ino.blocks <= 8);
+
+					let len = ino.size as usize;
+					let mut buf = vec![0u8; self.superblock.bsize as usize];
+					self.read_file_block(&ino, 0, &mut buf)?;
+					Ok(buf[0..len].to_vec())
```

Comment:
This clones the slice.  That's an extra allocation and data copy.  Better to truncate buf to len and then return buf.

Reply:
Thanks, I overlooked that.